JetMETCorrections/Modules/plugins/QGLikelihoodESProducer.cc

```diff
@@ -0,0 +1,58 @@
+/*
+ Description: ESProducer to get the quark-gluon likelihood object "QGLikelihoodObject"
+              from record "QGLikelihoodRcd". 
+
+ Implementation:
+     Completely trivial, simply returns the QGLikelihoodObject to the user. There is only
+     one QGLikelihoodObject object in each record. 
+*/
+// Original Author:  Salvatore Rappoccio
+//         Created:  Thu, 13 Mar 2014 15:02:39 GMT
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+
+#include "CondFormats/JetMETObjects/interface/QGLikelihoodObject.h"
+#include "CondFormats/DataRecord/interface/QGLikelihoodRcd.h"
+
+class QGLikelihoodESProducer : public edm::ESProducer {
+public:
+  QGLikelihoodESProducer(const edm::ParameterSet &);
+  ~QGLikelihoodESProducer() override{};
+
+  std::shared_ptr<const QGLikelihoodObject> produce(const QGLikelihoodRcd &);
+  void setIntervalFor(const edm::eventsetup::EventSetupRecordKey &, const edm::IOVSyncValue &, edm::ValidityInterval &);
+
+private:
+  edm::ESGetToken<QGLikelihoodObject, QGLikelihoodRcd> token_;
+};
+
+QGLikelihoodESProducer::QGLikelihoodESProducer(const edm::ParameterSet& iConfig) {
+  //the following line is needed to tell the framework what
+  // data is being produced
+  std::string label = iConfig.getParameter<std::string>("@module_label");
+  auto algo = iConfig.getParameter<std::string>("algo");
+  setWhatProduced(this, label).setConsumes(token_, edm::ESInputTag{"", algo});
+}
+
+// The same PDF's are valid for any time
+void QGLikelihoodESProducer::setIntervalFor(const edm::eventsetup::EventSetupRecordKey&,
+                                            const edm::IOVSyncValue&,
+                                            edm::ValidityInterval& oInterval) {
+  oInterval = edm::ValidityInterval(edm::IOVSyncValue::beginOfTime(), edm::IOVSyncValue::endOfTime());
+}
+
+// Produce the data
+std::shared_ptr<const QGLikelihoodObject> QGLikelihoodESProducer::produce(const QGLikelihoodRcd& iRecord) {
+  return std::shared_ptr<const QGLikelihoodObject>(&iRecord.get(token_), edm::do_nothing_deleter());
```

Comment:
Is the point of this class and QGLikelihoodSystematicsESProducer to act as an alias for the `QGLikelihood(Systematics)Object`? (i.e. provide a constant product label for the clients, while being able to have a single point where to make changes in case the product needs to point to something else in the DB(?))

Reply:
> Is the point of this class and QGLikelihoodSystematicsESProducer to act as an alias for the `QGLikelihood(Systematics)Object`? (i.e. provide a constant product label for the clients, while being able to have a single point where to make changes in case the product needs to point to something else in the DB(?))

@rappoccio may know the answer

Reply:
I would move forward with the integration of this PR, @santocch please have a look, leaving the clarification by @rappoccio as useful for possible further iterations